src/SchemaRegistry.ts

```diff
@@ -165,39 +166,50 @@ export default class SchemaRegistry {
   }
 
   private async updateOptionsWithSchemaReferences(
-    options: SchemaRegistryAPIClientOptions | undefined,
     schema: ConfluentSchema,
+    options: SchemaRegistryAPIClientOptions | undefined,
   ) {
     const helper = helperTypeFromSchemaType(schema.type)
-    const referredSchemas = await this.getReferredSchemas(schema, helper)
-    return helper.updateOptionsFromSchemaReferences(options as ProtocolOptions, referredSchemas)
+    const referencedSchemas = await this.getreferencedSchemas(schema, helper)
+
+    const protocolOptions = this.asProtocolOptions(options)
+    return helper.updateOptionsFromSchemaReferences(referencedSchemas, protocolOptions)
+  }
+
+  private asProtocolOptions(options?: SchemaRegistryAPIClientOptions): ProtocolOptions | undefined {
+    if (!(options as LegacyOptions)?.forSchemaOptions) {
+      return options as ProtocolOptions | undefined
+    }
+    return {
+      [SchemaType.AVRO]: (options as LegacyOptions)?.forSchemaOptions,
+    }
   }
 
-  private async getReferredSchemas(
+  private async getreferencedSchemas(
     schema: ConfluentSchema,
     helper: SchemaHelper,
   ): Promise<ConfluentSchema[]> {
     const referencesSet = new Set<string>()
-    return this.getReferredSchemasRecursive(schema, helper, referencesSet)
+    return this.getreferencedSchemasRecursive(schema, helper, referencesSet)
   }
 
-  private async getReferredSchemasRecursive(
+  private async getreferencedSchemasRecursive(
     schema: ConfluentSchema,
     helper: SchemaHelper,
     referencesSet: Set<string>,
   ): Promise<ConfluentSchema[]> {
     const references = schema.references || []
 
-    let referredSchemas: ConfluentSchema[] = []
+    let referencedSchemas: ConfluentSchema[] = []
     for (const reference of references) {
-      const schemas = await this.getReferredSchemasFromReference(reference, helper, referencesSet)
-      referredSchemas = referredSchemas.concat(schemas)
+      const schemas = await this.getreferencedSchemasFromReference(reference, helper, referencesSet)
+      referencedSchemas = referencedSchemas.concat(schemas)
```

Comment:
Why not just use `.push` and declare `referencedSchemas` a constant? It is also more safe, as one avoids accidentally directly mutating the declaration.

Reply:
It is because getreferencedSchemasFromReference returns a list

Reply:
You can resolve that with push too 🙂 Please notice [the documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push):

> The push() method adds one or more elements to the end of an array and returns the new length of the array.

```javascript
referencedSchemas.push(...schemas);
```